extras/find_imports.py

```diff
@@ -0,0 +1,136 @@
+# -*- coding: utf-8 -*-
+#
+# find_imports.py
+#
+# This file is part of NEST.
+#
+# Copyright (C) 2004 The NEST Initiative
+#
+# NEST is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# NEST is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+
+import glob
+import os
+import sys
+import re
+from collections import defaultdict
+
+
+"""Script to report all Python modules imported in NEST
+
+Run this script in the NEST source directory to obtain a
+listing of external modules imported by NEST and the part 
+importing them for those not part of the NEST core.
+
+This may be useful to define requirements.
+
+This is still work in progress and may return false positives.
+"""
+
+# directory to parse
+source_dir = '.'
```

Comment:
Most other checkers (e.g. [here](https://github.com/nest/nest-simulator/blob/master/extras/check_unused_names.py#L64), [here](https://github.com/nest/nest-simulator/blob/master/extras/check_copyright_headers.py#L58)) use `os.environ['NEST_SOURCE']` to get the path from the outside to make them independent of the working directory when calling. Can you do something similar here? If yes, please also update the docstring above.